tests/functional/common/vars-and-functions.sh

```diff
@@ -6,6 +6,15 @@ if [[ -z "${COMMON_VARS_AND_FUNCTIONS_SH_SOURCED-}" ]]; then
 
 COMMON_VARS_AND_FUNCTIONS_SH_SOURCED=1
 
+isTestOnNixOS() {
+  [[ "${isTestOnNixOS:-}" == 1 ]]
+}
+
+die() {
+  echo "fatal error: $*" >&2
+  exit 1
+}
```

Comment:
isn't this like `fail`?

Reply:
I wasn't aware, but it is _similar_. `fail` doesn't prefix anything.
They do seem to differ a bit in their use cases, which suggests that their behavior could be differentiated more. Perhaps:
- `die`: `echo "fatal error in test setup: $*"`
- `fail`: `echo "test failed: $*"`